### ISSUE TITLE
Why do u have 'setattr(self, 'fc%i' % i, fc)'

### ISSUE THREAD

**Issue by brando90** [MorvanZhou/PyTorch-Tutorial#19]
why do u have

    Why do u have 'setattr(self, 'fc%i' % i, fc)'

in ur code? is it really necessary? the standard cifar10 tutorial doesn't have it...

https://github.com/MorvanZhou/PyTorch-Tutorial/blob/master/tutorial-contents/504_batch_normalization.py

**Comment by brando90** [comment]
is:

    self._set_init(fc)

needed?

**Comment by MorvanZhou** [comment]
Here is the net without `setattr()`
···
Net(
  (bn_input): BatchNorm1d(1, eps=1e-05, momentum=0.5, affine=True)
  (bn0): BatchNorm1d(10, eps=1e-05, momentum=0.5, affine=True)
  (bn1): BatchNorm1d(10, eps=1e-05, momentum=0.5, affine=True)
  (bn2): BatchNorm1d(10, eps=1e-05, momentum=0.5, affine=True)
  (bn3): BatchNorm1d(10, eps=1e-05, momentum=0.5, affine=True)
  (bn4): BatchNorm1d(10, eps=1e-05, momentum=0.5, affine=True)
  (bn5): BatchNorm1d(10, eps=1e-05, momentum=0.5, affine=True)
  (bn6): BatchNorm1d(10, eps=1e-05, momentum=0.5, affine=True)
  (bn7): BatchNorm1d(10, eps=1e-05, momentum=0.5, affine=True)
  (predict): Linear(in_features=10, out_features=1)
)
···

Here is the net with that:
```
Net(
  (bn_input): BatchNorm1d(1, eps=1e-05, momentum=0.5, affine=True)
  (fc0): Linear(in_features=1, out_features=10)
  (bn0): BatchNorm1d(10, eps=1e-05, momentum=0.5, affine=True)
  (fc1): Linear(in_features=10, out_features=10)
  (bn1): BatchNorm1d(10, eps=1e-05, momentum=0.5, affine=True)
  (fc2): Linear(in_features=10, out_features=10)
  (bn2): BatchNorm1d(10, eps=1e-05, momentum=0.5, affine=True)
  (fc3): Linear(in_features=10, out_features=10)
  (bn3): BatchNorm1d(10, eps=1e-05, momentum=0.5, affine=True)
  (fc4): Linear(in_features=10, out_features=10)
  (bn4): BatchNorm1d(10, eps=1e-05, momentum=0.5, affine=True)
  (fc5): Linear(in_features=10, out_features=10)
  (bn5): BatchNorm1d(10, eps=1e-05, momentum=0.5, affine=True)
  (fc6): Linear(in_features=10, out_features=10)
  (bn6): BatchNorm1d(10, eps=1e-05, momentum=0.5, affine=True)
  (fc7): Linear(in_features=10, out_features=10)
  (bn7): BatchNorm1d(10, eps=1e-05, momentum=0.5, affine=True)
  (predict): Linear(in_features=10, out_features=1)
)
```

You now see the difference.
For your second question, my answer is we need `_set_init()` in this tutorial to witness how bad initialization could affect the result. But in your own application, you don't need it.
